### PR TITLE
Case 21060: Avatar Exporter v0.3.1 - index by actual joint names, store mappings separately

### PR DESCRIPTION
**Pull request by dback2**
https://highfidelity.manuscript.com/f/cases/21060

New Avatar Exporter package:
https://github.com/highfidelity/hifi/raw/44664ad7030a52159b759388a975b13952f6b1a0/tools/unity-avatar-exporter/avatarExporter.unitypackage

Do Not Merge:
The approach taken in this PR is just an idea right now and may not be desirable.

### CODE REVIEW

**Comment by dback2** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17247/HighFidelity-Beta-PR14911-953222a.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17247/HighFidelity-Beta-PR14911-953222a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17247/HighFidelity-Beta-Interface-PR14911-953222a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17247/HighFidelity-Beta-PR14911-953222a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17247/HighFidelity-Beta-Interface-PR14911-953222a.exe

File: libraries/model-baker/src/model-baker/PrepareJointsTask.cpp [hunk 1]
```diff
@@ -67,13 +69,18 @@ void PrepareJointsTask::run(const baker::BakeContextPointer& context, const Inpu
         auto jointNameMapKey = jointNameMapping.key(jointIn.name);
         if (jointNameMapping.contains(jointNameMapKey)) {
             jointOut.name = jointNameMapKey;
+        } else {
+            // If this FST prepended extra bone names with _ExtraBone then do the same here
+            QString extraBoneName = jointOut.name + "_ExtraBone";
+            if (offsets.contains(extraBoneName)) {
+                jointOut.name = extraBoneName;
+            }
```
Comment (huffman) [inline libraries/model-baker/src/model-baker/PrepareJointsTask.cpp]: Can we do all of this on the .fst side and add explicit `jointMap = Hip_ExtraBone = Hip` mappings?

Comment (amantley) [reply libraries/model-baker/src/model-baker/PrepareJointsTask.cpp]: I think I agree with Ryan.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17472/HighFidelity-Beta-PR14911-44664ad.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17472/HighFidelity-Beta-PR14911-44664ad.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17472/HighFidelity-Beta-Interface-PR14911-44664ad.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17472/HighFidelity-Beta-PR14911-44664ad.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17472/HighFidelity-Beta-Interface-PR14911-44664ad.exe

**Comment by dback2** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17481/HighFidelity-Beta-PR14911-44664ad.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17481/HighFidelity-Beta-Interface-PR14911-44664ad.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17481/HighFidelity-Beta-PR14911-44664ad.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17481/HighFidelity-Beta-Interface-PR14911-44664ad.exe

**Pull request closed without merging**